core/logger/zap_disk_logging.go

```diff
@@ -0,0 +1,95 @@
+package logger
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/utils"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+var diskPollInterval = 1 * time.Minute
+
+type zapDiskPollConfig struct {
+	stop     func()
+	pollChan <-chan time.Time
+}
+
+func newDiskPollConfig(interval time.Duration) zapDiskPollConfig {
+	ticker := time.NewTicker(utils.WithJitter(interval))
+
+	return zapDiskPollConfig{
+		pollChan: ticker.C,
+		stop:     ticker.Stop,
+	}
+}
+
+func newDiskCore(cfg ZapLoggerConfig) (zapcore.Core, error) {
+	availableSpace, err := cfg.diskStats.AvailableSpace(cfg.local.Dir)
+	if err != nil {
+		return nil, errors.Wrap(err, "error getting disk space available for logging")
+	}
+	if availableSpace < cfg.local.RequiredDiskSpace {
+		return nil, fmt.Errorf(
```

Comment:
Please always use `errors.Errorf` over `fmt.Errorf` everywhere since that bakes in useful debugging info like line numbers.